### DISCUSSION TITLE
Why Elsa?

### DISCUSSION THREAD

**Discussion by forresthopkinsa** [elsa-workflows/elsa-core#619] [Q&A] [answered]
Why is the project called Elsa? And is this documented anywhere?

It may seem like a silly question, but it's always one of the first answers I try to find when stumbling upon a project; and I haven't found it for this one. :)

**Comment by sfmskywalker** [comment, marked as answer]
I don't think I have it documented anywhere. And I don't think it's a silly question at all. The only silly questions out there are the ones not asked. :)

The answer, on the other hand, might be a different story. ;)

While I was trying to come up with a name for the project, I considered all things containing the word "workflow". And in a stroke of genius, I came up with "workflows core", since, you know, this being a .NET Core library and all. Needless to say, [a project with exactly that name](https://github.com/danielgerlag/workflow-core) already existed out there for years.

So I then tried to come up with plays on "workflows", like "flowy" and a whole sleeve of other names. I liked none of them because they were either too technical, too boring, too wordy, or already taken.

Finally I realized that maybe the name doesn't necessarily need convey the product itself, but instead could be abstract, inspirational or otherwise attractive as a brand. Like "Orchard". Who would have guessed that's an application framework? Or "Nancy", an MVC framework?

So I looked up names and aspects of rivers (since they "flow"). Turns out there's an Italian river named "Elsa". I immediately liked it because it was short and simple (and not used in the domain of workflows). It's also the name of an advanced AI in the Canadian tv-show "Travelers". The leap here is that if you squint real hard, a workflow diagram is a little bit like a connectome, as in that it consists of a set of nodes connected to one another. Not unlike an (artificial) brain. 

It might not make sense to anyone but me, but it stuck. Ultimately, it's just a handle to identify this project. 

So, there you have it: a documentation of the origins of the name for this project.

Thanks for asking!

**Reply by forresthopkinsa** [reply]
Fascinating! Thanks for taking the time to answer :)
